rclpy/src/rclpy/_rclpy.c

```diff
@@ -3448,23 +3451,34 @@ rclpy_get_node_names_and_namespaces(PyObject * Py_UNUSED(self), PyObject * args)
   }
   rcutils_string_array_t node_names = rcutils_get_zero_initialized_string_array();
   rcutils_string_array_t node_namespaces = rcutils_get_zero_initialized_string_array();
-  rcl_ret_t ret = rcl_get_node_names(node, allocator, &node_names, &node_namespaces);
+  rcutils_string_array_t security_contexts = rcutils_get_zero_initialized_string_array();
+  rcl_ret_t ret = RCL_RET_OK;
+  if (use_security_contexts) {
+    ret = rcl_get_node_names_with_security_contexts(
+      node, allocator, &node_names, &node_namespaces, &security_contexts);
+  } else {
+    ret = rcl_get_node_names(
+      node, allocator, &node_names, &node_namespaces);
+  }
   if (ret != RCL_RET_OK) {
     PyErr_Format(
-      RCLError, "Failed to get_node_names: %s", rcl_get_error_string().str);
+      RCLError, "Failed to get node names: %s", rcl_get_error_string().str);
     rcl_reset_error();
     return NULL;
   }
 
   rcutils_ret_t fini_names_ret;
   rcutils_ret_t fini_namespaces_ret;
+  rcutils_ret_t fini_security_contexts_ret;
   PyObject * pynode_names_and_namespaces = PyList_New(node_names.size);
   if (!pynode_names_and_namespaces) {
     goto cleanup;
   }
+
+  size_t tuple_size = use_security_contexts ? 3 : 2;
   size_t idx;
   for (idx = 0; idx < node_names.size; ++idx) {
-    PyObject * pytuple = PyTuple_New(2);
+    PyObject * pytuple = PyTuple_New(tuple_size);
```

Comment:
@ivanpauno nit^N: any reason not to keep everything in one line as in:
```suggestion
    PyObject * pytuple = PyTuple_New(use_security_context ? 3 : 2);
```
?

Reply:
Done